coriolis/utils.py

```diff
@@ -606,3 +609,198 @@ def create_service(ssh, cmdline, svcname, run_as=None, start=True):
     else:
         raise exception.CoriolisException(
             "could not determine init system")
+        
+        
+class Grub2ConfigEditor(object):
+    """This class edits GRUB2 configs, normally found in
+    /etc/default/grub. This class tries to preserve commented
+    and empty lines.
+    NOTE: This class does not actually write to file during
+    commit. Rhather, it will mutate it's internal view iof the
+    contents of that file with the latest changes made.
+    Use dump() to get the file contents.
+    """
+    def __init__(self, cfg):
+        self._cfg = cfg
+        self._parsed = self._parse_cfg(self._cfg)
+
+    def _parse_cfg(self, cfg):
+        ret = []
+        for line in cfg.split("\n")[:-1]:
```

Comment:
NIT: can we add a `if not line: continue` at the top to not run redundant checks in case of multiple newlines

Reply:
I wanted to retain the structure of the file as much as possible. That means comments and newlines, as the maintainers of that system intended that file to be. That is the purpose of the ```raw``` type.  The result is that the only thing that changes, are the values of the options you are mutating. The rest of the file remains the same (spaces, typos, comments, etc).